internal/services/postgres/postgresql_flexible_server_resource.go

```diff
@@ -1091,7 +1091,7 @@ func flattenFlexibleServerAuthConfig(ac *servers.AuthConfig) interface{} {
 }
 
 func expandFlexibleServerDataEncryption(input []interface{}) *servers.DataEncryption {
-	if len(input) == 0 {
+	if len(input) == 0 || input[0] == nil {
```

Comment:
@neil-yechenwei an empty `customer_managed_key` block isn't valid - so rather than doing this `key_vault_key_id` should become Required, since that's always going to be required for a customer managed key - can we update this?

Reply:
Sure. Updated.